src/shared/components/exonsCharts/ExonsChartStore.ts

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2018. The Hyve and respective contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * See the file LICENSE in the root of this repository.
+ *
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+import { computed } from "mobx";
+import { CancerStudy, Gene } from '../../api/generated/CBioPortalAPI';
+import { PfamDomain } from '../../api/generated/GenomeNexusAPI';
+import ResultViewFusionMapperDataStore from '../../../pages/resultsView/fusion/ResultViewFusionMapperDataStore';
+import {
+    EnsemblTranscriptExt, UtrType
+} from '../../model/Fusion';
+import { ExonRangeExt, PfamDomainRangeExt, StructuralVariantExt } from '../../model/Fusion';
+
+export class ExonsChartStore {
+
+    constructor(public gene: Gene,
+                public fusionDataStore: ResultViewFusionMapperDataStore,
+                public transcripts: EnsemblTranscriptExt[],
+                public study: { [studyId: string]: CancerStudy },
+                public pfamDomains: PfamDomain[],
+                public isLoadingEnsemblTranscripts: boolean) {
+    }
+
+    getExonsBySite(siteId: number, transcriptId: string, breakpoint: number): Array<ExonRangeExt> {
+        let exons: Array<ExonRangeExt> = [];
+        if (siteId === 1 || siteId === 2) {
+            let _t = this.getTranscriptById(transcriptId)[0];
+            let _sortedExons = <ExonRangeExt[]> _t.exons.sort(
+                // sort exons by rank
+                (exon1: ExonRangeExt, exon2: ExonRangeExt) => exon1.rank - exon2.rank
+            );
+            exons =
+                siteId < 2
```

Comment:
you're inside `siteId === 1 || siteId === 2`, so i think this condition should be `siteId === 1`